inference-engine/src/vpu/common/include/vpu/parsed_config_base.hpp

```diff
@@ -110,6 +110,14 @@ class ParsedConfigBase {
         return std::stoi(src);
     }
 
+    static int parsePositiveOrDefaultInt(const std::string& src) {
```

Comment:
Could you please explain function naming? I don't see an argument with default value here.
`orDefault` usually means that function does not throw an exception in case of unexpected input, but returns default in such cases

Reply:
@gladilov-gleb Maybe, I need to describe this idea.
[There is](https://github.com/openvinotoolkit/openvino/blob/7e057c522935f30760831f17c0e36b03a49e1ec7/inference-engine/tests/functional/plugin/myriad/shared_tests_instances/behavior/config.cpp#L80) a test to check for an invalid value:  
```
    INSTANTIATE_TEST_CASE_P(smoke_BehaviorTests, IncorrectConfigTests,
                            ::testing::Combine(
                                    ::testing::ValuesIn(netPrecisions),
                                    ::testing::Values(CommonTestUtils::DEVICE_MYRIAD),
                                    ::testing::ValuesIn(inconfigs)),
                            IncorrectConfigTests::getTestCaseName);
```  
Before this PR compile options were checked only in `CompileEnv::init`, not in `ParsedConfig::parse` so `IncorrectConfigTests` test can't pass.
But at the same time compile options have the "default" value `-1`.  For example, when they are not set, we call methods of `DefaultAllocation` to calculate an actual value.  

```
    int numSHAVEs = -1;
    int numCMXSlices = -1;
    int numExecutors = -1;
    int tilingCMXLimitKB = -1;
```  

`const auto numSlices  = config.numCMXSlices != -1 ? config.numCMXSlices : DefaultAllocation::numSlices(platform, numExecutors);`

So I added a method to check if the value is positive or not set (have default or auto value). This way I can add a test for an invalid negative value.

If you agree, do you have any suggestions to name this method?

Reply:
I think this way there shouldn't be such a function. Just check on _default_ before call already existing `positiveInt`.

If you have concerns about duplicating checks for the same _default_ value I guess can introduce local helper on caller side.